### ISSUE TITLE
Format source code using 4 character spacing

### ISSUE THREAD

**Issue by hazendaz** [dblock/oshi#90]
Use build-tools to do this with preset eclipse formatter.  This is a one time hit and will change many formatted items.  It should be done in a single commit with no other changes.

**Comment by dbwiddis** [comment]
PR #91 addresses. Please review and merge so we can see if build-tools makes any changes that Eclipse will keep reverting in diffs....

**Comment by hazendaz** [comment]
What do you mean by eclipse changing with diffs?  I know depending on how things are checked out to windows and jgit works it will sometimes look like line endings changed.  Do you mean what eclipse will do with bar-bone formatting?  That certainly would change things but the formatter enforces upon build.  The key though is to remember to build before committing.  I've been doing this for years and I'm still tripping myself up on making commits prior to running a build.  Build tools does just contain a formatter so certainly can just use it natively and go from there.  I ended up using it in waffle only because it was multiple module and having the empty file concept allowed for some cleanup in the poms.

**Comment by dbwiddis** [comment]
I want to configure my Eclipse formatter to use whatever coding conventions the build tool uses.   Up until now we've been using "Eclipse Default".  I gather there will be more changes (more brackets, etc.) so I'll want a set of coding convention xml files I can import into eclipse so my commits match the same conventions already in the code. 

I'm okay with any conventions, I just want them in Eclipse.

**Comment by dbwiddis** [comment]
And yes, I can do "mvn build" before committing but so much better if it reformats when I save, and I don't need to worry about it

**Comment by hazendaz** [comment]
Makes perfect sense.  Sonar generally will complain the most.  Checkstyles does too but I tend to ignore some of those complaints but you can see on my PR #92 that I did add some periods on javadocs.  That is where checkstyles complains.  I'll delete the formatter back off my PR so your cut can then come in with it the way you are doing it.  Then we can see what else the copy I have in build tools will cause.

Only note on this formatter.  It's based on luna at the moment.  There wasn't much in the way of mars additions and there are regressions in mars in some areas.  I'm working on the formatter project but its not in the best state for the mars version.  Basically we have been working to bring in a fork which was about 3 years on a different path but has m2e connector and javadoc formatting.  So maybe later in a few months that will become stable and get released.

**Comment by dbwiddis** [comment]
Mars completely rewrote the formatter and fixed a lot of shortcomings in Luna and previous versions.  I can switch back to Luna formatting in eclipse just as easily, that's not a problem.  I just want things to match and allow any other collaborators to have a quick way of importing settings into eclipse to match as well.

Keep the formatter on your pr if it's a file I can import into eclipse!

**Comment by hazendaz** [comment]
I had it as my build tools so the format.xml file was actually empty so I just removed it.  I can test with it after to see if it actually adds any tangable benefit.  It probably only does with multi module or if we wanted to use the google formatter which is also part of it.  It has two configurations to choose from googles or mine.  I haven't had enough time to really see what googles does so this could provide an opportunity to do so.  Either way, this being a single module project we can store the formatter xml file directly.

**Comment by dbwiddis** [comment]
OK, so I switched my eclipse to the default settings using the old formatter, and there are only 4 minor changes (whitespace, line breaks) from the current version.  So it's reasonable to assume the plugin is using the "eclipse default" settings as well.  And it may be using this format file:
https://github.com/revelc/formatter-maven-plugin/blob/master/src/config/eclipse/formatter/java.xml

Of note, that file includes this instruction:
`<setting id="org.eclipse.jdt.core.formatter.tabulation.char" value="tab"/>`

And our code still has tabs, not spaces.

So, I'm guessing if I change my current eclipse defaults to use spaces, and save the file as format.xml in the root directory, the plugin will use that instead?  Do we need the whole file or just that line to override?

**Comment by dbwiddis** [comment]
Reopened #91

**Comment by hazendaz** [comment]
Yes the whole file is needed.  On my person one I've got yet one more setting that eclipse forgot to export on luna.  Possibly fixed on mars but nor sure.  The file certainly can be imported but with its forceful nature that is not really required.  That is sort of the beauty in it.  A developer doesn't have to change their style but the plugin will enforce it.


Sent by Outlook for Android





On Wed, Sep 23, 2015 at 10:22 PM -0700, "dbwiddis" <notifications@github.com> wrote:
OK, so I switched my eclipse to the default settings using the old formatter, and there are only 4 minor changes (whitespace, line breaks) from the current version.  So it's reasonable to assume the plugin is using the "eclipse default" settings as well.  And it may be using this format file:
https://github.com/revelc/formatter-maven-plugin/blob/master/src/config/eclipse/formatter/java.xml

Of note, that file includes this instruction:
`<setting id="org.eclipse.jdt.core.formatter.tabulation.char" value="tab"/>`

And our code still has tabs, not spaces.

So, I'm guessing if I change my current eclipse defaults to use spaces, and save the file as format.xml in the root directory, the plugin will use that instead?  Do we need the whole file or just that line to override?

---
